src/compile/legend/encode.ts

```diff
@@ -81,9 +81,9 @@ export function gradient(fieldDef: FieldDef<string>, gradientSpec: any, model: U
   let gradient:any = {};
 
   if (type === 'gradient') {
-    const opacityDef = model.encoding.opacity;
-    if (isValueDef(opacityDef)) {
-      gradient.opacity = {value: opacityDef.value};
+    const opacity = getOpacityValue(model.encoding.opacity);
+    if (opacity) { // only apply opacity if it is zero or undefined
```

Comment:
"if it is neither zero or undefined"

I think this is a typo that I made earlier :P